### DISCUSSION TITLE
Get (all) Notes API including favorites and lists Notes

### DISCUSSION THREAD

**Discussion by kevincador** [trakt/trakt-api#416] [Q&A]
Hi @rectifyer,

First of all, private Notes and other Notes improvements are awesome! I’m really looking forward to add Notes everywhere in Rippple 🙌

I’m just looking at the Get Notes API and I’m missing the Notes associated with a List Item or Notes associated with the Favorite list. That’s something I found was missing when I integrated Notes in my app and I worked around by adapting the UI but now that I see the improvement to the API for other kind of Notes I’m scratching my head to find a way to integrate all of them in a way that makes sense.

I would totally understand if it’s not possible because of caching or other data specific reason, no worries, I can still scratch my head a bit 😉

Again, thanks a lot for the continuous effort to make Trakt better and to keep opening API feature 🙏

Kevin

**Comment by rectifyer** [comment]
All list notes are returned in those endpoints directly. Favorites, watchlist, or personal lists all return notes that way. They are stored differently behind the scenes and I wasn't planning on including them in the overall notes endpoint (or on web). Does that still make sense to you?

**Reply by kevincador** [reply]
I just wanted to be able to get all the notes for one media item to be able to display all of them on the media detail page. But I can keep only the media notes. 

I was also asking because I'm planing to also have a list of all notes accessible from the profile. I understand I'll be able to only show media notes, collection, rating and history notes but not favorites and lists notes. I think users may ask themselves why it's not all notes.

**Reply by rectifyer** [reply]
Yeah, that's a good point and something I need to figure out too. So you'd ideally want all notes to just be returned in the /users/:id/notes endpoint? I store them in different ways behind the scenes, so I'll need to see if it's possible combine them or I might need to require a specific list type to be specified for list notes.

**Reply by kevincador** [reply]
Yes, I was thinking being able to have notes linked to a list item like this for that endpoint:
```
"attached_to": {
      "type": "list",
      "id": 12345
    },
```

A specific filter type would be good yes, but it means the « all » wouldn’t be all (thinking about the UI and website more than the api).

It would also be cool to be able to get all notes for a specific media item. It would give a list with all the attached_to for that media item. Otherwise I’m thinking of fetching all the notes and cache that but I don’t know how users are going to use Notes.
